File: 026_RemoveDuplicatesFromSortedArray.c

//15.41%
//O(n)
int removeDuplicates(int* nums, int numsSize) {
    if(!nums || 1 >= numsSize)
		return numsSize;
	
	int size = 1;
	int val = nums[0];
	for(int i = 1; i < numsSize; i++)
		if(nums[i] == val){
			continue;
		else{
			nums[size++] = nums[i];
			val = nums[i];
		}
	}
	return size;
}

File: 026_RemoveDuplicatesFromSortedArray.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        size =  len(nums)
        if size <= 1:
            return size
        
        val = nums[0]
        size = 0
        for num in nums[1:]:
            if num > nums[size]:
                size, nums[size] = size + 1, num
        
        return size + 1
            